tomviz/FxiWorkflowWidget.cxx

```diff
@@ -209,6 +216,98 @@ vtkStandardNewMacro(InteractorStyle)
             &Internal::onPreviewRangeEdited);
   }
 
+  void setupUI(OperatorPython* pythonOp)
+  {
+    if (!pythonOp) {
+      return;
+    }
+
+    // If the user added extra parameters, add them here
+    auto json = QJsonDocument::fromJson(pythonOp->JSONDescription().toLatin1());
+    if (json.isNull() || !json.isObject()) {
+      return;
+    }
+
+    DataSource* ds = nullptr;
+    if (pythonOp->hasChildDataSource()) {
+      ds = pythonOp->childDataSource();
+    } else {
+      ds = qobject_cast<DataSource*>(pythonOp->parent());
+    }
+
+    if (!ds) {
+      ds = ActiveObjects::instance().activeDataSource();
+    }
+
+    QJsonObject root = json.object();
+
+    // Get the parameters for the operator
+    QJsonValueRef parametersNode = root["parameters"];
+    if (parametersNode.isUndefined() || !parametersNode.isArray()) {
+      return;
+    }
+    auto parameters = parametersNode.toArray();
+
+    // Here is the list of parameters for which we already have widgets
```

Comment:
Why not let the method automatically generate the UI for all the possible parameters?
What would happen if the user customizes the operator and removes one of the expected parameters?

Reply:
Well, having expected parameters allows us to do more things to help out the user. For instance, the "Rotation Center" parameter is automatically set to match whatever test rotation center the user selects. And the "Slice Stop" parameter is automatically set to the max slice value, and it is not allowed to go above that. These widgets are currently defined in the UI file.

For the Rotate Align Widget, the user also gets visual feedback from the parameters they choose. And there is opportunity for more complex parameter selection designs if there are expected parameters.

So I think there is a place for expected parameters. But we could potentially redesign how they are done. Maybe make them invisible and not used if the user removes them from the description file?

For now, though, the user shouldn't remove the expected parameters from the description.json file. I don't know what will happen if they do. I guess we'd have to design for that use case, maybe in a separate PR.